design/service-discovery-images/ClientListener-OnSubscriptionMatched.svg

```diff
@@ -0,0 +1,29 @@
+@startuml
+title ClientListener::OnSubscriptionMatched
+start
+if (MATCHED_MATCHING) then (yes)
+  : publishers_.insert(remoteEndpointGuid);
+  if (pending_matches_.find(response_publisher_guid)) then (yes)
+    : complete_matches_[response_publisher_guid] = pending_matches_[response_publisher_guid];
+    : pending_matches_.erase(response_publisher_guid);
+    : complete_matches_count_.store(complete_matches_.size());
+  endif
+else (no)
+  if (complete_matches_.find(response_publisher_guid)) then (yes)
+    : pending_matches_[response_publisher_guid] = complete_matches_[response_publisher_guid];
+    : complete_matches_.erase(response_publisher_guid);
+    : complete_matches_count_.store(complete_matches_.size());

```

Comment:
@IkerLuengo I think `complete_matches_count_` should be decremented before `complete_matches_` is modified.

Reply:
Good catch. Also corrected on ClientPubListener::OnPublicationMatched.